### PR TITLE
Add Error Handling to ORTOA

### PR DESCRIPTION
**Pull request by jasondu7297**
This PR accomplishes:
- Add error handling
NOTE: Unsure what happen to spacing and formatting... Will have to run the clang-format script to fix it...

### CODE REVIEW

File: client/client.cpp [hunk 1]
```diff
@@ -30,7 +30,8 @@ class ClientHandler {
 	ClientHandler(std::string path) {
 		seed_data.open(path);
 		if (!seed_data.is_open()) {
-        std::cerr << "File failed to open." << std::endl;
+			std::string err = "Invalid path to seed data";
+    	throw std::invalid_argument(err);
```
Comment (adriandavila) [inline client/client.cpp:34]: Does it not let you construct the error string directly inside?
i.e. `std::invalid_argument("Invalid path to seed data")`

File: client/client.cpp [hunk 2]
```diff
@@ -40,36 +41,32 @@ class ClientHandler {
 		auto protocol = std::make_shared<TBinaryProtocol>(transport);
 		RPCClient client(protocol);
 
-		try {
-			transport->open();
+		transport->open();
```
Comment (adriandavila) [inline client/client.cpp:44]: Can this throw? If yes then might want to wrap in a try-catch if we want to handle the error

Comment (jasondu7297) [reply client/client.cpp:44]: Yep this error is handled and caught by main

File: client/client.cpp [hunk 3]
```diff
@@ -40,36 +41,32 @@ class ClientHandler {
...
+		}
 
-			transport->close();
-		} catch (TException& tx) {
-			std::cout << "ERROR: " << tx.what() << std::endl;
-  	}
+		transport->close();
 	}
 
 	void getAveLatency() {
```
Comment (adriandavila) [inline client/client.cpp:72]: I dislike that you used `Ave` instead of `Avg` for average lol

File: client/client.cpp [hunk 4]
```diff
@@ -82,13 +79,18 @@ int main(int argc, char *argv[]) {
...
-	if (argc >= 2){
-		client = ClientHandler(argv[1]);
-	}
-
-	client.run();
-	client.getAveLatency();
+	try {
+		ClientHandler client;
+		if (argc >= 2){
+			client = ClientHandler(argv[1]);
```
Comment (adriandavila) [inline client/client.cpp:85]: can you alias argv[1] so that it's more clear what it's expecting

File: errors/OECreationFailed.h [hunk 5]
```diff
@@ -0,0 +1,13 @@
+#ifndef OECREATION_H
```
Comment (adriandavila) [inline errors/OECreationFailed.h:1]: Surely we can just have one "errors" header that defines many error classes right?

**Review by adriandavila — APPROVED** [review]
LGTM, minor changes requested

File: client/client.cpp (hunk 3 above)
Comment (jasondu7297) [reply client/client.cpp:72]: LOL I dislike that you like `Avg` more than `Ave`

**Pull request merged**